neuralprophet/time_net.py

```diff
@@ -516,11 +668,47 @@ def forward(self, inputs):
                     * ``events`` (torch.Tensor, float), all event features, dims (batch, n_forecasts, n_features)
                     * ``regressors``(torch.Tensor, float), all regressor features, dims (batch, n_forecasts, n_features)
 
+            meta : dict, default=None
+                Metadata about the all the samples of the model input batch.
+
+                Contains the following:
+
+                Model Meta:
+                    * ``df_name`` (list, str), time series ID corresponding to each sample of the input batch.
+
+                Note
+                ----
+                The meta is sorted in the same way the inputs are sorted.
+
+                Note
+                ----
+                The default None value allows the forward method to be used without providing the meta argument.
+                This was designed to avoid issues with the library `lr_finder` https://github.com/davidtvs/pytorch-lr-finder
+                while having  ``config_trend.trend_global_local="local"``.
+                The turnaround consists on passing the same meta (dummy ID) to all the samples of the batch.
+                Internally, this is equivalent to use ``config_trend.trend_global_local="global"`` to find the optimal learning rate.
+
         Returns
         -------
             torch.Tensor
                 Forecast of dims (batch, n_forecasts)
         """
+
+        # Turnaround to avoid issues when the meta argument is None in trend_global_local = 'local' configuration
+        # I'm repeating code here, config_season being None brings problems.
+        if meta is None and self.config_trend.trend_global_local == "local":
+            name_id_dummy = locals()["self"].id_list[0]
+            meta = OrderedDict()
+            meta["df_name"] = [name_id_dummy for _ in range(inputs["time"].shape[0])]
+            meta = torch.tensor([self.id_dict[i] for i in meta["df_name"]])
+        elif self.config_season is None:
+            pass
+        elif meta is None and self.config_season.season_global_local == "local":
```

Comment:
We could combine this with the first if block in:
if (trend == local) or (season == local):
...

Reply:
The problem here is the season_config = None, which I saw it happening